tests/aws/services/lambda_/test_lambda.py

```diff
@@ -2601,6 +2646,156 @@ def test_lambda_handler_update(self, aws_client, create_lambda_function, snapsho
             "invoke_result_handler_two_postpublish", invoke_result_handler_two_postpublish
         )
 
+    # TODO: Fix this test by not stopping running invokes for function updates of $LATEST
+    @pytest.mark.skip(
+        reason="""Fails with 'Internal error while executing lambda' because
+                  the current implementation stops all running invokes upon update."""
+    )
+    @markers.aws.validated
+    def test_function_update_during_invoke(self, aws_client, create_lambda_function, snapshot):
+        function_name = f"test-function-{short_uid()}"
+        environment_v1 = {"Variables": {"FUNCTION_VARIANT": "variant-1"}}
+        create_lambda_function(
+            func_name=function_name,
+            handler_file=TEST_LAMBDA_PYTHON_S3_INTEGRATION_FUNCTION_VERSION,
+            runtime=Runtime.python3_12,
+            Environment=environment_v1,
+        )
+
+        errored = False
+
+        def _update_function():
+            nonlocal errored
+            try:
+                # Make it very likely that the invocation is being processed
+                # In LocalStack, Lambda image pulling could potentially affect the exact timing,
+                # but at least the invocation runtime startup should be in progress.
```

Comment:
I think this comment is misleading. Image pulling should have already taken place during the transition to Active. This is maybe just for the container creation?

Reply:
good catch. I updated the comment.
Acquiring the lock is actually pretty fast for invokes 🚀 Container creation happens later and does not matter because we always send invokes to environments we spawn on demand (i.e., no optimization step here).